### PR TITLE
fix: add velocity-aware socnav personal space

### LINKED ISSUE #1323: enhancement: account for velocity in socnavbench personal space objective
## Goal / Problem

Update the SocNavBench personal-space objective so velocity-aware behavior can be evaluated without changing existing benchmark semantics by default.

Current code in `third_party/socnavbench/objectives/personal_space_cost.py` orients the asymmetric Gaussian from heading only:

```python
# TODO actually account for velocity here
personal_space_cost[0, i] += asym_gauss_from_vel(...)
```

The issue is not just a mechanical TODO cleanup. Velocity-dependent personal space is a behavior-changing research knob, so it must be explicit, default-off, and validated before any benchmark-default change.

## Scope

In scope:
- Add a velocity-aware personal-space mode behind an explicit config flag or parameter.
- Keep current behavior as the default for existing configs and benchmark paths.
- Use real agent velocity when available; preserve the current heading-derived fallback when velocity is unavailable or the new mode is disabled.
- Add focused tests for default behavior, enabled velocity-aware behavior, zero/stationary velocity handling, and fallback behavior.
- Add a tiny coefficient-sweep or smoke command over fixed scenarios to record whether candidate coefficients behave sensibly before any future default-on proposal.

Out of scope:
- Changing existing benchmark defaults.
- Tuning a final coefficient for paper-facing claims.
- Overhauling the SocNavBench objective framework.
- Treating improved realism as proven without scenario evidence.

## Added Value Estimation

- Maintenance value: low to medium; resolves a documented TODO while keeping behavior explicit.
- Research value: medium; enables controlled evaluation of dynamic personal-space assumptions.
- Benchmark value: low until coefficient-smoke evidence shows a useful effect without regressions.
- Why now: this is a bounded way to make the TODO executable without silently changing benchmark semantics.

## Effort Estimation

- Rough estimate (hours): 4
- Best estimate (hours): 3
- Unknowns:
  - exact velocity field exposed by the upstream `AgentState` object,
  - coefficient range for the smoke sweep,
  - whether existing SocNavBench fixtures need adaptation.

## Complexity Estimation

- Implementation complexity: medium.
- Dependencies:
  - `third_party/socnavbench/objectives/personal_space_cost.py`
  - SocNavBench params under `third_party/socnavbench/params/`
  - focused tests or smoke harness around the objective.
- Maintainer decisions resolved:
  - The selected path is default-off config/research knob first, not default behavior change.

## Risk Assessment

- Behavior-change risk: high if default-on, low when default-off.
- Compatibility risk: medium because this touches vendored/third-party benchmark code.
- Rollout risk: low if existing configs remain unchanged.
- Mitigation:
  - preserve current default behavior,
  - test default-off equivalence,
  - gate enabled behavior behind config,
  - record coefficient-smoke evidence before proposing any default change.

## Affected Files

- `third_party/socnavbench/objectives/personal_space_cost.py`
- `third_party/socnavbench/params/central_params.py` or equivalent config surface
- `tests/` or `test_pygame/` only if a targeted objective test needs local placement
- `docs/context/` if the coefficient-smoke result produces reusable guidance

## Definition of Done

- [ ] Existing/default personal-space objective output is unchanged when the new mode is disabled.
- [ ] Velocity-aware mode uses available agent velocity and falls back safely when velocity is unavailable or near-zero.
- [ ] Tests cover default-off equivalence, enabled behavior, zero velocity, and unavailable velocity.
- [ ] A small coefficient-sweep or smoke command is documented with fixed inputs/scenarios.
- [ ] Any recommendation about future default behavior is recorded as research evidence, not as an implicit benchmark claim.

## Success Metrics

- Existing benchmark/config paths remain behavior-compatible by default.
- Enabled mode produces a measurable difference in the intended direction for moving agents.
- The smoke output gives maintainers enough evidence to decide whether a later default-on issue is worth creating.

## Validation / Testing

Required local checks:

```bash
uv run pytest <targeted personal-space objective tests> -q
uv run ruff check third_party/socnavbench/objectives/personal_space_cost.py <touched-tests>
```

Required evidence before PR handoff:

```bash
BASE_REF=origin/main scripts/dev/pr_ready_check.sh
```

Required research smoke:

```bash
# exact command to be added by the implementation if no suitable harness exists yet
uv run python <coefficient-smoke-script> --fixed-scenarios <small-fixed-set>
```

## Estimate Discussion

- Why these values were proposed: the code change is small, but the validation must prove default behavior is preserved and enabled behavior is meaningful.
- Uncertainty / confidence: medium confidence; the main uncertainty is the available velocity API in SocNavBench agent state.
- What evidence would move the estimate: if velocity is not exposed cleanly, effort increases; if the objective is easily unit-testable, effort decreases.

## Project Metadata

- Priority: Medium
- Effort (h): 4
- Reviewed: 2026-05-20

## Maintainer Decision

Decision on 2026-05-20: use the default-off config/research-knob path. Do not change existing benchmark defaults in this issue.

### PR DESCRIPTION
**Pull request by ll7**
## Summary
Adds a default-off SocNavBench personal-space mode that uses the non-ego agent's stored scalar speed when available, while preserving the previous heading-derived unit vector by default.

## Linked Issues
- Closes #1323

## What Changed
- Added `use_agent_velocity`, `agent_velocity_scale`, and `min_agent_speed` parameters to the SocNavBench personal-space objective config.
- Updated `PersonalSpaceCost` to use finite, non-stationary `AgentState.current_config.speed_nk1()` values when the new mode is enabled.
- Added direct objective tests for default-off equivalence, positive-speed behavior, zero-speed fallback, and unavailable-speed fallback.
- Added a synthetic validation smoke for the coefficient sweep and recorded the objective-level evidence in `docs/context/issue_1323_socnavbench_personal_space_velocity.md`.

## Why It Matters
- Added value: velocity-aware personal-space experiments can now use the pedestrian speed already present in SocNavBench state.
- Expected impact: no default benchmark behavior change; the new behavior is opt-in.
- Why this is worth merging now: the issue contract is narrow, tested, and documented without overclaiming planner-level performance.

## Validation / Proof
- `uv run pytest tests/test_socnavbench_personal_space_cost.py -q` — 4 passed.
- `uv run python scripts/validation/run_issue_1323_socnav_personal_space_velocity_smoke.py` — passed, wrote disposable local summaries under `output/validation/issue_1323_socnav_personal_space_velocity/latest/`.
- `uv run pytest tests/test_planner/test_socnav.py -q` — 15 passed, 1 skipped.
- `BASE_REF=origin/main scripts/dev/check_docs_proof_consistency_diff.sh` — passed for 6 changed files.
- `PYTEST_NUM_WORKERS=8 BASE_REF=origin/main scripts/dev/pr_ready_check.sh` before latest-main sync — 3779 passed, 11 skipped, 6 warnings.
- After merging latest `origin/main`: `PYTEST_NUM_WORKERS=8 BASE_REF=origin/main scripts/dev/pr_ready_check.sh` — 3782 passed, 10 skipped, 3 warnings.

## Risks / Rollout
- Compatibility risks: default behavior remains disabled and uses the old heading-derived unit vector.
- Failure modes: if speed is missing, zero, near-zero, or non-finite, enabled mode falls back to default behavior.
- Rollback or fallback plan: disable `use_agent_velocity` or revert the objective/parameter changes.

## Docs / Provenance
- Updated docs: `docs/context/issue_1323_socnavbench_personal_space_velocity.md`, `docs/context/README.md`.
- Relevant design or provenance notes: the validation smoke is objective-level evidence only, not a SocNavBench planner benchmark claim.
- Assumptions: `AgentState.current_config.speed_nk1()` is the available SocNavBench scalar-speed source; vector direction is derived from heading.

## Follow-Up Issues
- Deferred work: planner-level SocNavBench benchmark evidence, if maintainers want this mode promoted beyond objective-level experiments.
- Issues opened for follow-up: none.

## Reviewer Notes
- Please verify that keeping `use_agent_velocity=False` by default is the desired rollout boundary.
- The branch includes a clean merge from latest `origin/main` before the final readiness run.

### CODE REVIEW

File: third_party/socnavbench/objectives/personal_space_cost.py [hunk 1]
```diff
@@ -13,12 +15,45 @@ class PersonalSpaceCost(Objective):
...
+        speed = float(speed_values[-1])
+        min_speed = float(self.p.get("min_agent_speed", 1e-3))
+        if not np.isfinite(speed) or abs(speed) <= min_speed:
+            return heading_velocity
+
+        velocity_scale = float(self.p.get("agent_velocity_scale", 1.0))
+        return (
+            float(velocity_scale * speed * np.cos(theta)),
+            float(velocity_scale * speed * np.sin(theta)),
+        )
```
[the remark this answers is not in the document]
Comment (ll7) [reply third_party/socnavbench/objectives/personal_space_cost.py:52]: Partially fixed in 93284e14: the PR-specific params are now cached in `__init__`, so this path no longer re-reads DotMap values per agent. I intentionally kept `min_agent_speed` because issue #1323 explicitly requires the zero/stationary/near-zero fallback; `uv run --extra socnav pytest tests/test_socnavbench_personal_space_cost.py -q` and the issue smoke both pass on the new head.

File: third_party/socnavbench/objectives/personal_space_cost.py [hunk 2]
```diff
@@ -36,16 +71,16 @@ def evaluate_objective(
             # iterate through every non ego agent
             agents: dict[str, AgentState] = sim_state.get_all_agents()
 
-            for _, agent_vals in agents.items():
+            for agent_vals in agents.values():
                 agent_pos3 = agent_vals.get_pos3()  # (x,y,th)
                 theta = agent_pos3[2]
+                velx, vely = self._agent_velocity_from_config(agent_vals, theta)
```
[the remark this answers is not in the document]
Comment (ll7) [reply third_party/socnavbench/objectives/personal_space_cost.py:82]: Reviewed with a second pass. I am not changing the loop structure in this PR: the agent iteration inside the trajectory-point loop is pre-existing SocNavBench objective shape, and this PR only adds a cached velocity/scale lookup on that path. Local validation on 93284e14 covers the new behavior: personal-space unit tests, issue #1323 smoke, and ruff on the touched objective/test/smoke files.

**Comment by ll7** [comment]
Applied merge-ready after full-proof review.

Basis: #1323 scope is resolved by a default-off velocity-aware SocNavBench personal-space mode; review threads were handled/resolved; CI is green; PR body records targeted personal-space tests, SocNav planner tests, issue smoke, docs/proof consistency, and full post-main-sync readiness (`3782 passed, 10 skipped`). Generated validation outputs are classified as disposable under `output/validation/...`.
